cpp/src/arrow/csv/converter.cc

```diff
@@ -275,10 +282,29 @@ Status NumericConverter<T>::Convert(const BlockParser& parser, int32_t col_index
   StringConverter<T> converter;
 
   auto visit = [&](const uint8_t* data, uint32_t size, bool quoted) -> Status {
+    // XXX should quoted values be allowed at all?
     value_type value;
     if (IsNull(data, size, quoted)) {
       return builder.AppendNull();
```

Comment:
Even though it's not a part of the changeset: why do we use the safe method here, but an unsafe one at around 312? Couldn't it both be unsafe?

Reply:
Yes, good point.